src/model.rs

```diff
@@ -2566,6 +2592,179 @@ mod tests {
         }
     }
 
+    mod enhance_line_names {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn empty_line_with_non_empty_route() {
+            let mut collections = Collections::default();
+            collections
+                .lines
+                .push(Line {
+                    id: String::from("line_id1"),
+                    name: String::new(),
+                    ..Default::default()
+                })
+                .unwrap();
+            collections
+                .routes
+                .push(Route {
+                    id: String::from("route_id1"),
+                    line_id: String::from("line_id1"),
+                    direction_type: Some(String::from("forward")),
+                    name: String::from("my route id1"),
+                    ..Default::default()
+                })
+                .unwrap();
+
+            let lines_to_routes =
+                OneToMany::new(&collections.lines, &collections.routes, "lines_to_routes").unwrap();
+
+            collections.enhance_line_names(&lines_to_routes);
+            let line1 = collections.lines.get("line_id1").unwrap();
+            assert_eq!("my route id1", line1.name);
+        }
+
+        #[test]
+        fn non_empty_line_with_non_empty_route() {
+            let mut collections = Collections::default();
```

Comment:
If the line were handled by the `ModelBuilder` you can use it here and I think it would be nice to.
This could be written as 
```rust
let model = transit_model_builder::ModelBuilder::default()
        .line("line_id1", |l| {l.name = "my line id".to_owned();})
        .route("route_id1", |r| {
            r.line_id = "line_id1".to_owned();
            r.direction_type = Some("forward".to_owned());
            r.name = "my route id1".to_owned();
        })
        .build();
```

Reply:
I can add it to your PR if you want writing rust would be a nice Jenkins/kube break :grin:

Reply:
we keep it in mind for later :smile: